### PR TITLE
Pal 1081 update example lib

### PR DESCRIPTION
**Pull request by m78233**

### CODE REVIEW

**Review by ac74475 — COMMENTED** [review]
I know the way you have gone about this keeps the jars to a minimum size, however I think doing so has increased the deployment complexity. I was expecting that there wouldn't need to be any changes to the deployment and services directories as new user and rule libraries would be included in the example-lib jar with dependancies.

**Comment by ac74475** [comment]
after discussions it looks like an equal tradeoff, so we will leave it as it is now

File: example-user-library/pom.xml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>uk.gov.gchq.palisade</groupId>
+        <artifactId>example</artifactId>
+        <version>0.5.0-${revision}</version>
```
Comment (developer01189998819991197253) [inline example-user-library/pom.xml:23]: These poms need updating to use 0.5.1

File: example-rule-library/pom.xml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>uk.gov.gchq.palisade</groupId>
+        <artifactId>example</artifactId>
+        <version>0.5.0-${revision}</version>
```
Comment (developer01189998819991197253) [inline example-rule-library/pom.xml:23]: These poms need updating to use 0.5.1

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: deployment-jvm/local-jvm/README.md [hunk 3]
```diff
@@ -87,10 +87,12 @@ The above steps can be automated using the provided scripts, all of which are in
      drwxrwxrwx Palisade-examples
    >> cd Palisade-examples
    >> ls
-     drwxrwxrwx deployment-k8s
      drwxrwxrwx deployment-jvm
+     drwxrwxrwx deployment-k8s
+     drwxrwxrwx example-library
      drwxrwxrwx example-library
      drwxrwxrwx example-runner
+     drwxrwxrwx example-library
```
[the remark this answers is not in the document]
Comment (m78233) [reply deployment-jvm/local-jvm/README.md:95]: Fixed, lists includes example-library, example-rule-library, example-user-library

File: example-library/src/test/resources/application-example-test.yaml [hunk 4]
```diff
@@ -74,4 +74,4 @@ population:
   serialisers:
   - flavourFormat: ${default.format}
     flavourType: ${default.type}
-    serialiserClass: uk.gov.gchq.palisade.example.library.serialise.AvroSerialiser
+    serialiserClass: uk.gov.gchq.palisade.example.seralise.AvroSerialiser
```
[the remark this answers is not in the document]
Comment (m78233) [reply example-library/src/test/resources/application-example-test.yaml:77]: done

File: example-rule-library/pom.xml (hunk 2 above)
Comment (m78233) [reply example-rule-library/pom.xml:23]: Gave it a go and found I was hitting-up against issues with the dependencies that are not with the right dependencies.  I'm thinking that we need to do this in sequence of the dependencies, i.e. common, service, reader, client and then examples.

**Review by dev930018 — APPROVED** [review]

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**